romancal/tweakreg/tweakreg_step.py

```diff
@@ -134,460 +128,325 @@ def process(self, input):
-
-        # initial shift to be used with absolute astrometry
-        self.abs_xoffset = 0
-        self.abs_yoffset = 0
-
-        self.abs_refcat = self.abs_refcat.strip()
-        gaia_cat_name = self.abs_refcat.upper()
-
-        if gaia_cat_name in SINGLE_GROUP_REFCAT:
-            with images:
-                models = list(images)
-
-                try:
-                    # FIXME: astrometric_utils expects all models in memory
-                    ref_cat = amutils.create_astrometric_catalog(
-                        models,
-                        gaia_cat_name,
-                        output=output_name,
+                    image_model.meta["tweakreg_catalog"] = catalog.as_array()
+                    nsources = len(catalog)
+                    self.log.info(
+                        f"Detected {nsources} sources in {image_model.meta.filename}."
+                        if nsources
+                        else f"No sources found in {image_model.meta.filename}."
                     )
-                except Exception as e:
-                    self.log.warning(
-                        "TweakRegStep cannot proceed because of an error that "
-                        "occurred while fetching data from the VO server. "
-                        f"Returned error message: '{e}'"
+                    # build image catalog
+                    # catalog name
+                    catalog_name = os.path.splitext(image_model.meta.filename)[0].strip(
+                        "_- "
                     )
-                    self.log.warning("Skipping 'TweakRegStep'...")
-                    self.skip = True
-                    for model in models:
-                        model.meta.cal_step["tweakreg"] = "SKIPPED"
-                    [images.shelve(m, i, modify=False) for i, m in enumerate(models)]
-                    return images
-                [images.shelve(m, i, modify=False) for i, m in enumerate(models)]
-
-        elif os.path.isfile(self.abs_refcat):
-            ref_cat = Table.read(self.abs_refcat)
-
-        else:
-            raise ValueError(
-                "'abs_refcat' must be a path to an "
-                "existing file name or one of the supported "
-                f"reference catalogs: {_SINGLE_GROUP_REFCAT_STR}."
-            )
+                    # catalog data
+                    catalog_table = Table(image_model.meta.tweakreg_catalog)
+                    catalog_table.meta["name"] = catalog_name
+
+                    imcats.append(
+                        {
+                            "model_index": i,
+                            "imcat": tweakreg.construct_wcs_corrector(
+                                wcs=image_model.meta.wcs,
+                                refang=image_model.meta.wcsinfo,
+                                catalog=catalog_table,
+                                group_id=image_model.meta.group_id,
+                            ),
+                        }
+                    )
```

Comment:
```suggestion
                    imcat = tweakreg.construct_wcs_corrector(
                                wcs=image_model.meta.wcs,
                                refang=image_model.meta.wcsinfo,
                                catalog=catalog_table,
                                group_id=image_model.meta.group_id,
                            )
                    imcat.meta["model_index"] = i
                    imcats.append(imcat)
```
The correctors have a `meta` attribute (sort of like datamodels) that can be used here to store the `model_index`.